homeassistant/components/aurora_abb_powerone/config_flow.py

```diff
@@ -56,55 +89,109 @@ def validate_and_connect(
     return ret
 
 
-def scan_comports() -> tuple[list[str] | None, str | None]:
-    """Find and store available com ports for the GUI dropdown."""
-    com_ports = serial.tools.list_ports.comports(include_links=True)
-    com_ports_list = []
-    for port in com_ports:
-        com_ports_list.append(port.device)
-        _LOGGER.debug("COM port option: %s", port.device)
-    if len(com_ports_list) > 0:
-        return com_ports_list, com_ports_list[0]
-    _LOGGER.warning("No com ports found.  Need a valid RS485 device to communicate")
-    return None, None
+def __validate_and_connect_tcp(hass: core.HomeAssistant, data):
+    host = data[CONF_HOST]
+    port = data[CONF_PORT]
+    address = data[CONF_ADDRESS]
+    _LOGGER.debug("Intitialising connection to %s:%s [%s]", host, port, address)
+    ret = {}
+    ret["title"] = DEFAULT_INTEGRATION_TITLE
+    try:
+        client = AuroraTCPClient(host, port, address, timeout=1)
+        client.connect()
+        ret[ATTR_SERIAL_NUMBER] = client.serial_number()
+        ret[ATTR_MODEL] = f"{client.version()} ({client.pn()})"
+        ret[ATTR_FIRMWARE] = client.firmware(1)
+        _LOGGER.info("Returning device info=%s", ret)
+    except AuroraError as err:
+        _LOGGER.warning("Could not connect to device %s:%s [%s]", host, port, address)
+        raise err
+    finally:
+        client.close()
+
+    # Return info we want to store in the config entry.
+    return ret
 
 
 class AuroraABBConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Aurora ABB PowerOne."""
 
     VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+    _comportslist = None
```

Comment:
Is there a reason for moving these here rather than leaving them defined in `__init__`?